### PR TITLE
chore: add tests for the preset

### PR DESCRIPTION
**Pull request by JakeChampion**

### CODE REVIEW

File: src/jest-preset.ts [hunk 1]
```diff
@@ -19,7 +19,6 @@ export default {
     fastly,
   },
   moduleNameMapper: {
-    "^fastly:.*":
-      "node_modules/jest-preset-fastly-js-compute/fastly-js-compute-mock.js",
+    "^fastly:.*": join(__dirname, "./fastly-js-compute-mock.js"),
```
Comment (JakeChampion) [inline src/jest-preset.ts:22]: This change allows us to test the preset locally

File: src/fastly-js-compute-mock.ts [hunk 2]
```diff
@@ -26,7 +26,7 @@ export class Backend {
   }
 
   public toString(): string {
-    return JSON.stringify(this.config, null, "  ");
+    return this.config.name;
```
Comment (JakeChampion) [inline src/fastly-js-compute-mock.ts:29]: This fixes a bug in the mock implementation to match the real implementation

**Comment by JakeChampion** [comment]
@ysugimoto this is also ready for review ☺️

File: src/jest-preset.ts (hunk 1 above)
Comment (ysugimoto) [reply src/jest-preset.ts:22]: Looks a good idea!

File: src/fastly-js-compute-mock.ts [hunk 3]
```diff
@@ -234,7 +234,7 @@ export class KVStore {
   }
 
   put(key: string, value: BodyInit): Promise<undefined> {
-    this.store[key] = new KVStoreItem(value);
+    this.store[key] = new KVStoreEntry(value);
```
Comment (ysugimoto) [inline src/fastly-js-compute-mock.ts:237]: 👍

File: jest.config.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    "^.+\\.tsx?$": [
+      "ts-jest",
+      {
+        useESM: true,
+      },
+    ],
+  },
+};
+
+export default jestConfig;
```
Comment (ysugimoto) [inline jest.config.ts:16]: Question

Is this config used for the actual testing? I find you remove the `defaultESM` at jest-preset.ts then I'm wondering if underlying ESM support could work or not

**Review by ysugimoto — COMMENTED** [review]
I've just put a question for treating ESM transpiration but looks entirely nice, thanks!

File: src/typescript/esm/jest-preset.ts [hunk 5]
```diff
@@ -1,6 +1,5 @@
 /// <reference types="@fastly/js-compute" />
 
-import defaultESM from "ts-jest/presets/default-esm/jest-preset";
 import {
```
Comment (JakeChampion) [inline src/typescript/esm/jest-preset.ts:3]:
```suggestion
import defaultESM from "ts-jest/presets/default-esm/jest-preset";
import {
```

File: src/typescript/esm/jest-preset.ts [hunk 6]
```diff
@@ -9,9 +8,8 @@ import {
   fetch,
   fastly,
 } from "../../globals";
-
+import { join } from "node:path";
 export default {
-  ...defaultESM,
   globals: {
```
Comment (JakeChampion) [inline src/typescript/esm/jest-preset.ts:13]:
```suggestion
  ...defaultESM,
  globals: {
```

File: jest.config.ts (hunk 4 above)
Comment (JakeChampion) [reply jest.config.ts:16]: I have re-added `...defaultESM` 👍

**Comment by ysugimoto** [comment]
Thanks! This PR could be helpful for future improvement.

**Review by ysugimoto — APPROVED** [review]

**Pull request merged**